### DISCUSSION TITLE
How can I send realtime updates without writing to a DDS? (signals)

### DISCUSSION THREAD

**Discussion by tylerbutler** [microsoft/FluidFramework#5469] [Documentation topics]
This article isn't written yet, but you can discuss and vote on it here. We prioritize writing articles that receive the most votes.

---

Draft info: https://github.com/microsoft/FluidFramework/tree/main/packages/runtime/datastore-definitions#signals

Sending signals: [submitSignal](https://fluidframework.com/docs/apis/datastore-definitions/ifluiddatastoreruntime/#datastore-definitions-ifluiddatastoreruntime-submitsignal-MethodSignature)

```ts
IFluidDataStoreRuntime.submitSignal(type: string, content: any): void;
```

To listen to signals, use the `"signal"` event:

```ts
 dataObject1.runtime.on("signal", (message: IInboundSignalMessage, local: boolean) => {
                if (message.type === "MySignal") {
                    // do something
                }
            });
```

**Comment by scaryrawr** [comment]
@tylerbutler are there any examples that make use of signals?

**Reply by tylerbutler** [reply]
There are uses within the framework itself but not in the examples at the moment. That said, I added some draft info to the discussion here that should point you in the right direction.

**Comment by gund** [comment]
IMO signals are useful in case you need to represent user presence on UI - like a live position of the cursors in the editor.
So you would send signals of every user's cursor position and then render them dynamically in the editor.
